src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +60,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
     _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (!_isRunning.compareAndSet(false, true)) {
```

Comment:
It seem unnecessary to still use AtomaticBoolean if these methods are synchronzied. Can you change `_isRunning` to `volatile boolean` everywhere? Thanks.